arangod/RocksDBEngine/RocksDBBuilderIndex.cpp

```diff
@@ -167,24 +442,80 @@ Result RocksDBBuilderIndex::remove(transaction::Methods& trx,
   return Result();  // do nothing
 }
 
+static Result processPartitions(
+    bool isForeground, std::deque<std::pair<uint64_t, uint64_t>> partitions,
+    trx::BuilderTrx& trx, rocksdb::Snapshot const* snap,
+    RocksDBCollection* rcoll, rocksdb::DB* rootDB, RocksDBIndex& ridx,
+    std::atomic<uint64_t>& docsProcessed, size_t numThreads,
+    uint64_t threadBatchSize, rocksdb::Options const& dbOptions,
+    std::string const& idxPath) {
+  auto sharedWorkEnv =
+      std::make_shared<SharedWorkEnv>(std::move(partitions), rcoll->objectId());
+  std::vector<std::unique_ptr<IndexCreatorThread>> idxCreatorThreads;
+  for (size_t i = 0; i < numThreads; ++i) {
+    auto newThread = std::make_unique<IndexCreatorThread>(
+        false, isForeground, threadBatchSize, docsProcessed, sharedWorkEnv,
+        rcoll, rootDB, ridx, snap, dbOptions, idxPath);
+    idxCreatorThreads.emplace_back(std::move(newThread));
+  }
+
+  try {
+    for (auto& idxCreatorThread : idxCreatorThreads) {
+      idxCreatorThread->start();
+    }
+  } catch (std::exception const& ex) {
+    LOG_TOPIC("01ad6", WARN, Logger::ENGINES)
+        << "error while starting index creation thread: " << ex.what();
+    // abort the startup
+    sharedWorkEnv->registerError({TRI_ERROR_INTERNAL, ex.what()});
+  }
+  sharedWorkEnv->waitUntilAllThreadsTerminate();
+
+#ifdef ARANGODB_ENABLE_MAINTAINER_MODE
+  uint64_t seekCounter = 2;
+  uint64_t nextCounter = 0;
+  for (auto const& threadStats : sharedWorkEnv->getThreadStatistics()) {
+    seekCounter += threadStats.numSeeks;
+    nextCounter += threadStats.numNexts;
+  }
+  LOG_DEVEL << "Total seeks: " << seekCounter << ", next: " << nextCounter;
```

Comment:
We shouldn't LOG_DEVEL the statistics here.
Instead, let's remove all the `#ifdef ARANGODB_ENABLE_MAINTAINER_MODE` things for the statistics, so they are recorded unconditionally.
And here, we can log the statistics using a proper `LOG_TOPIC`, but give it a `DEBUG` log severity. That means the statistics will normally not be contained in the log files, but if one turns on debug logging, they will be logged. We also need to adjust the log message so it gets a bit more context (for example, it doesn't say that it is related to an index creation etc.).